File: testData/src/java8/pkg/TestTryCatchThrowable.java
package pkg;

public class TestTryCatchThrowable {

  void testEmptyCatch(){
    try {
      System.out.println("Hi");
    } catch (Throwable ignored){
    }
  }

  void testEmptyCatchWithTail(){
    try {
      System.out.println("Hi");
    } catch (Throwable ignored){
    }

    System.out.println("tail");
  }

  String testCatchWithReturn(){
    try {
      System.out.println("Hi");
      return "5";
    } catch (Throwable ignored){
      return null;
    }
  }

  String testCatchWithReturnAndTail(){
    try {
      System.out.println("Hi");
    } catch (Throwable ignored){
      return null;
    }
    System.out.println("Ho");
    return "bye";
  }

  String testCatchNested(){
    try {
      System.out.println("Hi");
      return "hello";
    } catch (Throwable ignored){
      try {
        System.out.println("Hi!");
        return "5";
      } catch (Throwable throwable){
        return throwable.getMessage();
      }
    }
  }

  String testCatchWithRethrow(){
    try {
      System.out.println("Hi");
      return "5";
    } catch (Throwable throwable){
      System.out.println("Oh no");
      throw throwable;
    }
  }


  void testCatchAndOtherEmptyCatch(){
    try {
      System.out.println("Hi");
    } catch (RuntimeException ignored) {

    } catch (Throwable ignored){
      System.out.println("Oh no");
    }
  }

  void testCatchAndOtherNonEmptyCatch(){
    try {
      System.out.println("Hi");
    } catch (RuntimeException ignored) {
      System.out.println("Hello");
    } catch (Throwable ignored){
      System.out.println("Oh no");
    }
  }
}

File: plugins/kotlin/testData/src/kt/pkg/TestTryCatchThrowable.kt
package pkg

class TestTryCatchThrowable {
    fun testEmptyCatch() {
        try {
            println("Hi")
        } catch (ignored: Throwable) {
        }
    }

    fun testEmptyCatchWithTail() {
        try {
            println("Hi")
        } catch (ignored: Throwable) {
        }

        println("tail")
    }

    fun testCatchWithReturn(): String? {
        try {
            println("Hi")
            return "5"
        } catch (ignored: Throwable) {
            return null
        }
    }

    fun testCatchWithReturnAndTail(): String? {
        try {
            println("Hi")
        } catch (ignored: Throwable) {
            return null
        }
        println("Ho")
        return "bye"
    }

    fun testCatchNested(): String? {
        try {
            println("Hi")
            return "hello"
        } catch (ignored: Throwable) {
            try {
                println("Hi!")
                return "5"
            } catch (throwable: Throwable) {
                return throwable.message
            }
        }
    }

    fun testCatchWithRethrow(): String {
        try {
            println("Hi")
            return "5"
        } catch (throwable: Throwable) {
            println("Oh no")
            throw throwable
        }
    }


    fun testCatchAndOtherEmptyCatch() {
        try {
            println("Hi")
        } catch (ignored: RuntimeException) {
        } catch (ignored: Throwable) {
            println("Oh no")
        }
    }

    fun testCatchAndOtherNonEmptyCatch() {
        try {
            println("Hi")
        } catch (ignored: RuntimeException) {
            println("Hello")
        } catch (ignored: Throwable) {
            println("Oh no")
        }
    }
}

